README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->



## Features

Simple text styles can be implemented instantly.
Used Material Design3 Typography.
-> https://m3.material.io/styles/typography/type-scale-tokens

Basic Properties:
* display
* headline
* title
* body
* caption
* button
* overline

## Getting started

In your flutter project add the dependency:
```
dependencies: 
  instant_text_styles: ^<version>
```
Import package:
```dart
import 'package:instant_text_styles/instant_text_styles.dart';
```

## Usage

The simplest example:
```dart
Text('Title Large', style: TextStyles.title.large),
```

Example of changing colors:
```dart
Text('Title Large Black', style: TextStyles.title.black),
Text('Title Large White', style: TextStyles.title.white),
```

Example of changing color according to status:
```dart
//Color(0xFF34C759)
Text('Success!!', style: TextStyles.body.success),

//Color(0xFFFF3B30)
Text('Error!!', style: TextStyles.body.error),

//Color(0xFFFF9500)
Text('Warning!!', style: TextStyles.body.warning),
```

Example of text decoration:
```dart
// Draw a line underneath each line of text
Text('Underline', style: TextStyles.body.underline),

// Draw a line through each line of text
Text('LineThrough', style: TextStyles.body.lineThrough),

// Draw a line above each line of text
Text('Overline', style: TextStyles.body.overline),
```

If you want to adjust parameters:

```dart
Text('Use copyWith', style: TextStyles.body.copyWith(color: Colors.amber)),
```


CHANGELOG.md
## 0.0.1+1

* Add homepage url.
## 0.0.1

* Initial release.


lib/src/text_styles.dart
import 'package:flutter/material.dart';

///* Used Material Design3
///
/// Use as:
///
/// Text("Hello!", styles: TextStyles.title.large.white);
///
class TextStyles {
  TextStyles._();

  // static const fontFamily = "Roboto";

  static _toStyle(double fontSize, double height, FontWeight fontWeight,
      double letterSpacing,
      {Color? color, TextDecoration? decoration}) {
    return TextStyle(
      // fontFamily: TextStyles.fontFamily,
      color: color,
      fontSize: fontSize,
      // textBaseline: TextBaseline,
      // baseline 設定がうまくいかず height 指定すると下寄りの文字になってしまうので、適用を保留
      height: height / fontSize,
      fontWeight: fontWeight,
      letterSpacing: letterSpacing,
      decoration: decoration,
    );
  }


  ///* large: _toStyle(size: 57, height: 64, FontWeight.w500, spacing: 0),
  ///
  ///* medium: _toStyle(size: 45, height:52, FontWeight.w500, spacing: 0),
  ///
  ///* small: _toStyle(size: 36, height:44, FontWeight.w500, spacing: 0),
  static final display = _MyTextStyleSet(
    large: _toStyle(57, 64, FontWeight.w400, 0),
    medium: _toStyle(45, 52, FontWeight.w400, 0),
    small: _toStyle(36, 44, FontWeight.w400, 0),
  );

  ///* large: _toStyle(size: 32, height:40, FontWeight.w500, spacing: 0),
  ///
  ///* medium: _toStyle(size: 28, height:36, FontWeight.w500, spacing: 0),
  ///
  ///* small: _toStyle(size: 24, height:32, FontWeight.w500, spacing: 0),
  static final headline = _MyTextStyleSet(
    large: _toStyle(32, 40, FontWeight.w400, 0),
    medium: _toStyle(28, 36, FontWeight.w400, 0),
    small: _toStyle(24, 32, FontWeight.w400, 0),
  );

  ///* large: _toStyle(size: 22, height:28, FontWeight.w500, spacing: 0),
  ///
  ///* medium: _toStyle(size: 16, height:24, FontWeight.w500, spacing: 0.15),
  ///
  ///* small: _toStyle(size: 14, height:20, FontWeight.w500, spacing: 0.1),
  static final title = _MyTextStyleSet(
    large: _toStyle(22, 28, FontWeight.w400, 0),
    medium: _toStyle(16, 24, FontWeight.w500, 0.15),
    small: _toStyle(14, 20, FontWeight.w500, 0.1),
  );

  ///* large: _toStyle(size: 14, height:20, FontWeight.w500, spacing: 0.1),
  ///
  ///* medium: _toStyle(size: 12, height:16, FontWeight.w500, spacing: 0.5),
  ///
  ///* small: _toStyle(size: 11, height:16, FontWeight.w500, spacing: 0.5),
  static final label = _MyTextStyleSet(
    large: _toStyle(14, 20, FontWeight.w500, 0.1),
    medium: _toStyle(12, 16, FontWeight.w500, 0.5),
    small: _toStyle(11, 16, FontWeight.w500, 0.5),
  );

  ///* large: _toStyle(size: 16, height:24, FontWeight.w400, spacing: 0.15),
  ///
  ///* medium: _toStyle(size: 14, height:20, FontWeight.w400, spacing: 0.25),
  ///
  ///* small: _toStyle(size: 12, height:16, FontWeight.w500, spacing: 0.4),
  static final body = _MyTextStyleSet(
    large: _toStyle(16, 24, FontWeight.w400, 0.15),
    medium: _toStyle(14, 20, FontWeight.w400, 0.25),
    small: _toStyle(12, 16, FontWeight.w400, 0.4),
  );

  ///* (size: 12, height: 16, FontWeight.w500, spacing: 0.4),
  static final caption = body.small;

  ///* (size: 14, height: 20, FontWeight.w500, spacing: 0.1),
  static final button = label.large;

  ///* (size: 11, height: 16, FontWeight.w500, spacing: 0.4),
  static final overline = label.small;
}

/// Wrapper for TextStyle
class _MyTextStyle extends TextStyle {
  _MyTextStyle(TextStyle style)
      : super(
          color: style.color,
          backgroundColor: style.backgroundColor,
          fontSize: style.fontSize,
          fontWeight: style.fontWeight,
          fontStyle: style.fontStyle,
          letterSpacing: style.letterSpacing,
          wordSpacing: style.wordSpacing,
          textBaseline: style.textBaseline,
          height: style.height,
          leadingDistribution: style.leadingDistribution,
          locale: style.locale,
          foreground: style.foreground,
          background: style.background,
          shadows: style.shadows,
          fontFeatures: style.fontFeatures,
          decoration: style.decoration,
          decorationColor: style.decorationColor,
          decorationStyle: style.decorationStyle,
          decorationThickness: style.decorationThickness,
          debugLabel: style.debugLabel,
          overflow: style.overflow,
        );

  _MyTextStyle get black =>
      _MyTextStyle(copyWith(color: const Color(0xFF000000)));

  _MyTextStyle get white =>
      _MyTextStyle(copyWith(color: const Color(0xFFFFFFFF)));

  _MyTextStyle get grey =>
      _MyTextStyle(copyWith(color: const Color(0xFF8E8E93)));

  _MyTextStyle get success =>
      _MyTextStyle(copyWith(color: const Color(0xFF34C759)));


  _MyTextStyle get error =>
      _MyTextStyle(copyWith(color: const Color(0xFFFF3B30)));

  _MyTextStyle get warning =>
      _MyTextStyle(copyWith(color: const Color(0xFFFF9500)));

  _MyTextStyle get link => _MyTextStyle(copyWith(color: Colors.blueAccent));

  /// FontWeight.w700
  _MyTextStyle get bold => _MyTextStyle(copyWith(fontWeight: FontWeight.bold));

  /// FontWeight.w600
  _MyTextStyle get medium =>
      _MyTextStyle(copyWith(fontWeight: FontWeight.w600));

  /// FontWeight.w400
  _MyTextStyle get light => _MyTextStyle(copyWith(fontWeight: FontWeight.w400));

  _MyTextStyle get underline =>
      _MyTextStyle(copyWith(decoration: TextDecoration.underline));

  _MyTextStyle get lineThrough =>
      _MyTextStyle(copyWith(decoration: TextDecoration.lineThrough));

  _MyTextStyle get overline =>
      _MyTextStyle(copyWith(decoration: TextDecoration.overline));
}

class _MyTextStyleSet extends _MyTextStyle {
  final _MyTextStyle large;
  // medium is this
  final _MyTextStyle small;

  _MyTextStyleSet(
      {required TextStyle large,
      required TextStyle medium,
      required TextStyle small})
      : large = _MyTextStyle(large),
        small = _MyTextStyle(small),
        super(medium);
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:instant_text_styles/instant_text_styles.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'TextStyles Example',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    const gap = SizedBox(height: 16);
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text('Title Large 日本語フォント', style: TextStyles.title.large),
              Text('Title Medium', style: TextStyles.title),
              Text('Title Small', style: TextStyles.title.small),
              gap,
              Text('Label Large', style: TextStyles.label.large),
              Text('Label Medium', style: TextStyles.label),
              Text('Label Small', style: TextStyles.label.small),
              gap,
              Text('Body Large', style: TextStyles.body.large),
              Text('Body Medium', style: TextStyles.body),
              Text('Body Small', style: TextStyles.body.small),
              gap,
              //Color(0xFF34C759)
              Text('Success!!', style: TextStyles.body.success),
              //Color(0xFFFF3B30)
              Text('Error!!', style: TextStyles.body.error),
              //Color(0xFFFF9500)
              Text('Warning!!', style: TextStyles.body.warning),
              gap,
              Text('Underline', style: TextStyles.body.underline),
              Text('LineThrough', style: TextStyles.body.lineThrough),
              Text('Overline', style: TextStyles.body.overline),
              gap,
              Text('Use copyWith',
                  style: TextStyles.body.copyWith(color: Colors.amber)),
            ],
          ),
        ),
      ),
    );
  }
}
